src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestParser.cs

```diff
@@ -0,0 +1,411 @@
+                {
+                    node.Add(ParseComment());
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return node;
+        }
+
+        private T ParseTrailingTrivia<T>(T node, bool stopAfterNewLine = false) where T : HttpSyntaxNode
+        {
+            while (MoreTokens())
+            {
+                if (stopAfterNewLine && CurrentToken.Kind is HttpTokenKind.NewLine)
+                {
+                    ConsumeCurrentTokenInto(node);
+                    break;
+                }
+
+                if (CurrentToken.Kind is not (HttpTokenKind.Whitespace or HttpTokenKind.NewLine))
+                {
+                    break;
+                }
+
+                ConsumeCurrentTokenInto(node);
+            }
+
+            return node;
+        }
+
+        private HttpRequestNode ParseRequest()
+        {
+            var methodNode = ParseMethod();
+            var urlNode = ParseUrl();
+            var versionNode = ParseVersion();
+            var headersNode = ParseHeaders();
+            var bodySeparatorNode = ParseBodySeparator();
+            var bodyNode = ParseBody();
+
+            return new HttpRequestNode(
+                _sourceText,
+                _syntaxTree,
+                methodNode,
+                urlNode,
+                versionNode,
+                headersNode,
+                bodySeparatorNode,
+                bodyNode);
+        }
+
+        private HttpMethodNode? ParseMethod()
+        {
+            var node = new HttpMethodNode(_sourceText, _syntaxTree);
+
+            ParseLeadingTrivia(node);
+
+            if (MoreTokens() && CurrentToken.Kind is HttpTokenKind.Word)
+            {
+                if (CurrentToken.Kind is HttpTokenKind.Word && CurrentToken.Text.ToLower() is ("get" or "post" or "patch" or "put" or "delete" or "head" or "options" or "trace"))
+                {                 
+                    ConsumeCurrentTokenInto(node);
+                }
+                else if (CurrentToken.Kind is HttpTokenKind.Word && CurrentToken.Text.ToLower() is not ("get" or "post" or "patch" or "put" or "delete" or "head" or "options" or "trace" or "https"))
+                {        
+                    var tokenSpan = _sourceText.GetSubText(CurrentToken.Span).Lines.GetLinePositionSpan(CurrentToken.Span);                
+
+                    var diagnostic = new Diagnostic(LinePositionSpan.FromCodeAnalysisLinePositionSpan(tokenSpan), DiagnosticSeverity.Warning, CurrentToken.Text.ToLower(), $"Unrecognized HTTP verb {CurrentToken.Text}");
+                    node.AddDiagnostic(diagnostic);
+                    ConsumeCurrentTokenInto(node);
+                }
+                else if (CurrentToken.Kind is HttpTokenKind.Word && CurrentToken.Text.ToLower() is "https")
+                {
+                    return null;
```

Comment:
`null` returns should always happen in an early out block at the very top of the function (before we even construct the node). Otherwise, we will end up parsing the leading trivia onto the `node` variable which is discarded and never attached to the tree. The resulting tree would then fail to round trip (because it would no longer preserve the complete text in the original string).

```
private HttpMethodNode? ParseMethod()
{
    if (MoreTokens() && CurrentToken.Kind is HttpTokenKind.Word && CurrentToken.Text.ToLower() is "https")
    {
        return null;
    }
    
    var node = new HttpMethodNode(_sourceText, _syntaxTree);

    ParseLeadingTrivia(node);

...
}